### PR TITLE
Correctly determine lockfile path in helm repo add

### PR DESCRIPTION
**Pull request by xanecs**
**What this PR does / why we need it**:
`helm repo add` automatically creates a lockfile based on the repository config file path by replacing the file extension with `.lock`.
When the given filepath for `--repository-config` did not include a file extension, a lockfile in a nonexistent directory
would have been created.
For Example `helm --repository-config /tmp/repo-config repo add ...` resulted in Helm trying to create `.lock/tmp/repo-config`.
This PR corrects this, so that the extension is appended in those cases.

**Special notes for your reviewer**:

**If applicable**:
- [ ] this PR contains documentation
- [x] this PR contains unit tests
- [ ] this PR has been tested for backwards compatibility

### CODE REVIEW

**Comment by bacongobbler** [comment]
Please write a test case to cover this new code path. Thanks

**Review by bacongobbler — APPROVED** [review]
Code LGTM! Thank you for implementing those test cases.

This will need another maintainer's approval before we can merge. Thanks!

**Review by hickeyma — APPROVED** [review]
LGTM, thanks for fixing @xanecs

**Pull request merged**
